resources/packs/gcp/monitoring.go

```diff
@@ -0,0 +1,193 @@
+package gcp
+
+import (
+	"context"
+	"fmt"
+
+	kms "cloud.google.com/go/kms/apiv1"
+	monitoring "cloud.google.com/go/monitoring/apiv3/v2"
+	monitoringpb "cloud.google.com/go/monitoring/apiv3/v2/monitoringpb"
+	"go.mondoo.com/cnquery/llx"
+	"go.mondoo.com/cnquery/resources"
+	"go.mondoo.com/cnquery/resources/packs/core"
+	"google.golang.org/api/iterator"
+	"google.golang.org/api/option"
+)
+
+func (g *mqlGcpProjectMonitoringService) id() (string, error) {
+	projectId, err := g.ProjectId()
+	if err != nil {
+		return "", err
+	}
+	return fmt.Sprintf("%s/gcp.project.monitoringService", projectId), nil
+}
+
+func (g *mqlGcpProjectMonitoringService) init(args *resources.Args) (*resources.Args, GcpProjectMonitoringService, error) {
+	if len(*args) > 0 {
+		return args, nil, nil
+	}
+
+	provider, err := gcpProvider(g.MotorRuntime.Motor.Provider)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	projectId := provider.ResourceID()
+	(*args)["projectId"] = projectId
+
+	return args, nil, nil
+}
+
+func (g *mqlGcpProject) GetMonitoring() (interface{}, error) {
+	projectId, err := g.Id()
```

Comment:
Is this correct?

L18 uses another function to get the projectID:
`g.Id()` vs. `g.ProjectId()`

Reply:
Those are 2 different resources. On L18 we are working with `mqlGcpProjectMonitoringService` that has a property `projectId`. Here we are working with a `mqlGcpProject` that has `id`.